### DISCUSSION TITLE
Lingering On LED

### DISCUSSION THREAD

**Discussion by SalvadorZXA** [gpiozero/gpiozero#1143] [Q&A]
Hi, I have an LED that is turned on in the script, but it stays that way even if the script is terminated/closed/exited. I presume it's suppose to turn off at closing?

**API Version:** 2.0.1

**Solution Attempts:**
1) Using "atexit" and manually setting them off using ".off()" and/or ".close()"
2) Trying to use "signal" and "SIGINT" with "SIGTERM" with ".off()" and/or ".close()"

**Sample Code:**
```
from gpiozero import Button, LED

LEDStatus = False

whiteLED = LED(5)
whiteLEDButton = Button(6, pull_up=True, bounce_time=0.1)

def callWhiteLED():
    global LEDStatus
    
    if LEDStatus:
        whiteLED.off()
        LEDStatus = False
    else:
        whiteLED.on()
        LEDStatus = True

whiteLEDButton.when_pressed = callWhiteLED
```

**Comment by bennuttall** [comment]
Do you have the same problem simply using

```python
from gpiozero import LED

led = LED(5)
led.on()
```

How are you exiting/terminating?

**Reply by SalvadorZXA** [reply]
Yes

**Reply by bennuttall** [reply]
How are you running/exiting/terminating?

**Reply by SalvadorZXA** [reply]
> How are you running/exiting/terminating?

CTRL + D or pressing the "X" on the top right window of the running script.

**Reply by bennuttall** [reply]
gpiozero performs a clean-up when the script ends, which restores the pins to the state they were at when the program started. Was the LED on when you started?

Are you using Raspberry Pi OS?

**Reply by SalvadorZXA** [reply]
> gpiozero performs a clean-up when the script ends, which restores the pins to the state they were at when the program started. Was the LED on when you started?
> 
> Are you using Raspberry Pi OS?

LED is off on boot and remained that way till I specified the code to turn it on as seen above. Just for fun I even tried leaving it on intentionally, re-running the script while on, but it would still not turn it off at close/termination.

**OS:** Raspberry Pi OS (64-bit)

**Additional Info:** I'm using a Raspberry Pi 5. I've tried this code on 2 pies and 3 different SDs.

**Reply by lurch** [reply]
Can you please try running `pinctrl -v get 5` immediately after your Pi has booted, and also after running your gpiozero script?
